tool-plugins/vscode/plugin/debugger/index.js

```diff
@@ -230,7 +231,7 @@ class BallerinaDebugSession extends LoggingDebugSession {
             scope = "Global";
             frameId = frameId - 1000;
         }
-        const frame = this.debugArgs.frames[frameId];
+        const frame = this.debugArgs[this.currentThreadId].frames[frameId];
```

Comment:
Ah. so not considering the thereadid was why the variables did not show up. But the thread id should be taken from `args` param of `variablesRequest`. Maybe it'll come up when there are multiple workers?

Could you try with this program?

import ballerina/io;
```ballerina
function main(string... args) {
    worker w1 {
        int workerid = 1
        io:println("workerid", workerid);  //<-- breakpoint here
    }
    worker w2 {
        int workerid = 2
        io:println("workerid", workerid);  //<-- breakpoint here
    }
}
```

If the workerid variable value shows correctly for both workers then this works. But I think they both will show the same number.

Reply:
Like you said workerId shows incorrectly here.
`args.thereadid` is undefined how do we add attributes to args of variablesRequest?